ui/src/app/components/app-router.tsx

```diff
@@ -0,0 +1,46 @@
+import * as fp from 'lodash/fp';
+import * as React from 'react';
+import { BrowserRouter, Link, Redirect, Route, Switch, useHistory, useLocation, useParams, useRouteMatch} from 'react-router-dom';
+
+const {Fragment} = React;
+
+interface Guards {
+  [index: number]: {
+    checkGuard: () => boolean;
+    redirectPath: string;
+  };
+}
+
+export const usePath = () => {
+  const {path} = useRouteMatch();
+  return path;
+};
+
+export const SubRoute = ({children}): React.ReactElement => <Switch>{children}</Switch>;
+export const AppRouter = ({children}): React.ReactElement => <BrowserRouter><SubRoute>{children}</SubRoute></BrowserRouter>;
+
+export const RouteLink = ({path, style = {}, children}): React.ReactElement => <Link style={{...style}} to={path}>{children}</Link>;
+
+export const AppRoute = ({path, data = {}, component: Component}): React.ReactElement => {
+  const routeParams = useParams();
+  const routeHistory = useHistory();
+
+  return <Route path={path} >
+    <Component urlParams={routeParams} routeHistory={routeHistory} routeConfig={data}/>
+  </Route>;
+};
+
+export const ProtectedRoutes = (
+  {path, guards, children}: {path: string, guards: Guards, children: any[] }): React.ReactElement => {
```

Comment:
can children be more strongly typed here?  Fragment[] perhaps?

Reply:
I did mean to update this - I set it to React.ReactNode. That should cover any type of child that react accepts